2019-Q4-PyPI/RFI.md

```diff
@@ -0,0 +1,164 @@
+
+## Scope
+
+This Request for Information is seeking **software developers** to design, implement, test, verify, and assist in the rollout of the following features to the codebase that powers [PyPI](https://pypi.org/).
+
+- Verifiable cryptographic signing of artifacts ([PEP 458](https://www.python.org/dev/peps/pep-0458/)/[TUF](https://theupdateframework.github.io) or simiilar)
+- Technical infrastructure and methods for automated detection of malicious package uploads
+- Continuation of high priority improvements to previous security work on API Tokens and Multi-Factor authentication
+- Documentation for above features
+
+### Open Questions
+
+#### Verifiable Cryptographic Signing of Artifacts
+
+[PEP 458](https://www.python.org/dev/peps/pep-0458/) and [PEP 480](https://www.python.org/dev/peps/pep-0480/) were authored to propose implementation of [The Update Framework](https://theupdateframework.github.io) on [PyPI](https://pypi.org/) to allow end users to verify the integrity of packages downloaded from PyPI.
+
+We are hoping that this RFI will allow the community, experts, and potential contractors to explore prior art in other package repositories, and discuss potential solutions for PyPI. This feedback will allow the PSF to decide what solution to request in our Request for Proposals.
+
+Key discussion points are:
+
+- Acceptance criteria for the community
+- Implementation feasibility for PyPI
+- Future maintainability and support requirements from PyPI administrators
+- Security and operational efficacy of proposed solutions
+
+#### Automated Detection of Malicious Uploads
+
+[PyPI](https://pypi.org/) adds tens of thousands of new releases across the projects hosted in the repository and thousands of new projects monthly.
+
+There are regular ongoing attempts by bad actors to upload releases and artifacts that include malicious payloads either in [`setup.py` files](https://www.bleepingcomputer.com/news/security/python-package-installation-can-trigger-malicious-code/) or within the package contents itself. Additionally, spam and scam artists sometimes attempt to create projects that include references and links to deceive search indexes and users.
+
+The moderation capacity of the PyPI team is limited and currently relies on reports from the community to address malicious package uploads and spam/scam postings on the site.
+
+We are hoping that this RFI will allow the community, experts, and potential contractors to explore prior art in other package repositories, and discuss potential solutions for PyPI. This feedback will allow the PSF to decide what solution to request in our Request for Proposals.
+
+Key discussion points are:
+
+- Acceptance criteria for the PyPI maintainers and moderators
+- Methods for detection of malicious content
+- Implementation feasibilty for PyPI and associated scalability
+
+## Estimated Budgets and Caps
```

Comment:
This section feels confusing -- I'd like for us to go ahead and provide estimates and caps in the RFI stage, but if we're not going to do that, I'd like to explicitly say that "**Budgets and caps** will be provided in the RFP to help contractors in preparing their proposals" to clarify expectations.